akka-stream-typed/src/main/scala/akka/stream/typed/javadsl/ActorFlow.scala

```diff
@@ -64,6 +64,20 @@ object ActorFlow {
         JavaDurationConverters.asFiniteDuration(timeout))
       .asJava
 
+  /**
+   * Use for messages whose response is known to be a [[akka.pattern.StatusReply]]. When a [[akka.pattern.StatusReply#success]] response
+   * arrives the future is completed with the wrapped value, if a [[akka.pattern.StatusReply#error]] arrives the future is instead
+   * failed.
+   */
+  def askWithStatus[I, Q, A](
+      ref: ActorRef[Q],
+      timeout: java.time.Duration,
+      makeMessage: BiFunction[I, ActorRef[StatusReply[A]], Q]): Flow[I, A, NotUsed] =
+    akka.stream.typed.scaladsl.ActorFlow
+      .askWithStatus[I, Q, A](parallelism = 2)(ref)((i, ref) => makeMessage(i, ref))(
```

Comment:
Could this be passing the method reference for `BiFunction.apply` instead of creating an additional lambda that only calls that method (same for Scala api)?